### PR TITLE
add command to migrate legacy backups to current format

### PR DESCRIPTION
**Pull request by skriss**
Signed-off-by: Steve Kriss <krisss@vmware.com>

@carlisia @nrb FYI this is what my migration command looks like.  I won't actually be merging this into the head of master -- I'll have to create a `release-0.11` branch and PR against that.

Testing so far looks good, but I would like to run through one end-to-end scenario starting with a v0.9 install and upgrading to v0.11.

Bikeshedding on the command name welcome.

### CODE REVIEW

**Comment by skriss** [comment]
Sample output:

```
2019/04/11 16:21:20 + PREPARING
2019/04/11 16:21:20    + getting Velero API client
2019/04/11 16:21:20    + getting backup storage location "default"
2019/04/11 16:21:21    + discovering plugins
2019/04/11 16:21:21    + creating plugin manager
2019/04/11 16:21:21    + getting object store plugin
2019/04/11 16:21:21    + initializing object store plugin
2019/04/11 16:21:21    + getting backup store
2019/04/11 16:21:21    + listing backups in backup store
2019/04/11 16:21:21 + PROCESSING BACKUPS
2019/04/11 16:21:21    + processing backup "v0.9"
2019/04/11 16:21:21       + checking whether backup is in the current format
2019/04/11 16:21:21       + starting migration of backup to current format
2019/04/11 16:21:21       + getting ark-backup.json metadata file from object storage
2019/04/11 16:21:21       + processing backup.status.volumeBackups
2019/04/11 16:21:21          + processing backup.status.volumeBackups[pvc-9f52e22c-5b13-11e9-a2bc-06b65eb65ade]
2019/04/11 16:21:21          + processing backup.status.volumeBackups[pvc-a22656d6-5b13-11e9-a2bc-06b65eb65ade]
2019/04/11 16:21:21       + encoding volume snapshots as v0.9-volumesnapshots.json.gz
2019/04/11 16:21:21       + uploading v0.9-volumesnapshots.json.gz to object storage
2019/04/11 16:21:21       + encoding backup metadata to velero-backup.json
2019/04/11 16:21:21       + uploading velero-backup.json to object storage
2019/04/11 16:21:21       + deleting ark-backup.json from object storage
2019/04/11 16:21:21       + migration of backup complete!
2019/04/11 16:21:21 + MIGRATION SUMMARY
2019/04/11 16:21:21    + total backups processed:       1
2019/04/11 16:21:21    + already current backups:       0
2019/04/11 16:21:21    + successful migrations:         1
2019/04/11 16:21:21    + failed migrations:             0
```

**Comment by skriss** [comment]
Ran through a full end-to-end test - started with a v0.9 install, created a backup, then upgraded to v0.10 followed by v0.11, then ran this command, then confirmed the backup was properly rewritten, and it was restorable. One small bug fix that affects the `describe` output, which I'll push a fix for in a sec.

**Comment by nrb** [comment]
`velero migrate-backups` seems a little bit more clear to me, especially since the output uses `migration`.

**Comment by skriss** [comment]
@carlisia @nrb rebased onto `release-0.11`, ready for review.

File: pkg/cmd/cli/upgradebackups/upgrade_backups.go [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+	"github.com/heptio/velero/pkg/plugin"
+	"github.com/heptio/velero/pkg/util/encode"
+	"github.com/heptio/velero/pkg/volume"
+)
+
+/*
+
+// upgrade backups in the "default" backup location, and any snapshots
+// found will be written as being in the "aws-default" snapshot location
+velero upgrade --backup-location default --volume-snapshot-location aws-default
```
Comment (nrb) [inline pkg/cmd/cli/upgradebackups/upgrade_backups.go]: Should this go into the `Example` field on the command?

Comment (skriss) [reply pkg/cmd/cli/upgradebackups/upgrade_backups.go]: Oh, yeah, I think this was just a note for myself before I started coding. I think `Example` already captures this. Will remove.

**Comment by skriss** [comment]
I'll also rename the command and package to `migrate-backups`.

**Comment by nrb** [comment]
Will merge once the name change is in.

**Comment by skriss** [comment]
Updated name to `migrate-backups`, added changelog, and squashed.

**Pull request merged**